packages/core/src/graph/graph.ts

```diff
@@ -1265,16 +1267,95 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
     });
 
     //
-    // 2. Add the items to the graph
+    // 2. Compute the origin positions
+    //
+    const existingNodesMap = this.getNodes().reduce((total, node) => {
```

Comment:
Could we use `this.get('itemMap')` ?
[source](https://github.com/sirensolutions/G6/blob/51e841c250f11b79edea046d9741b4fc80fde8e0/packages/core/src/graph/graph.ts#L1015)